apps/web-client/src/lib/db/cr-sqlite/warehouse.ts

```diff
@@ -1,7 +1,35 @@
-import type { PickPartial } from "@librocco/db";
+/**
+ * @fileoverview Warehouse management system
+ *
+ * Warehouse Overview:
+ * - A warehouse represents a logical grouping of books based on categories or attributes
+ * - Examples: "New Books", "Used Books", "2025 School Year", "Computer Science", etc.
+ * - Each warehouse can have a display name and an optional discount that applies to all of its books
+ * - Books are moved in to or out of warehouses via "notes" - these are transactions that increase/decrease a batch of isbn-quantities
+ * - Book counts are maintained through a running total of these transactions
+ * - Warehouses are created with auto-incrementing display names (e.g., "New Warehouse", "New Warehouse (2)")
+ *
+ * Data Sources:
+ * - warehouse table:
+ *   - id: Primary key
+ *   - display_name: Human readable name
+ *   - discount: Optional percentage discount applied to books
+ * - note table: Contains metadata about transactions
```

Comment:
We don't list individual fields elsewhere in the "Data sources" sections, so we should not do it here.

```suggestion
 * - warehouse table:
 * - note table: Contains metadata about transactions
```